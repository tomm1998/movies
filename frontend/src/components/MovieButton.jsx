import React from "react";

class MovieButton extends React.Component {
  state = { isClicked: false };

  handleClick = (e) => {
    if (this.state.clicked) return;

    e.preventDefault();
    this.setState({ isClicked: true }, () => {
      setTimeout(() => {
        this.setState({ isClicked: false });
      }, 200);
    });
    this.props.onClick();
  };

  render() {
    let styles;
    let clickedStyles;
    if (this.props.primary) {
      styles = {
        color: "white",
        background: "#0532FF",
      };
    }
    if (this.state.isClicked) {
      clickedStyles = {
        opacity: "0.6",
      };
    }

    return (
      <div
        role="button"
        {...this.props}
        onClick={this.handleClick}
        style={{
          margin: "1rem",
          padding: "0.5rem",
          minWidth: "80px",
          borderRadius: "4px",
          display: "inline-block",
          textAlign: "center",
          textTransform: "uppercase",
          fontWeight: "bold",
          background: "#cdcdcd",
          ...styles,
          ...clickedStyles,
        }}
      >
        {this.props.label}
      </div>
    );
  }
}

export default MovieButton;
