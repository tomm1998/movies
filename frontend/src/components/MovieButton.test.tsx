import { render, fireEvent, screen } from "@testing-library/react";
import React from "react";
import { describe, it, expect, vi } from "vitest";
import MovieButton from "./MovieButton";

describe("Card", () => {
  it("renders", () => {
    const wrapper = render(<MovieButton label="label" />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it("does handle click", () => {
    const mock = vi.fn();
    expect(mock).not.toHaveBeenCalledOnce();
    render(<MovieButton onClick={mock} label="label" />);

    fireEvent.click(screen.getByText("label"));
    expect(mock).toHaveBeenCalledOnce();
  });
});
