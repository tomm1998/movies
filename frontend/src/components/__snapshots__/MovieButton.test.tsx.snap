// Vitest Snapshot v1

exports[`Card > renders 1`] = `
<div>
  <div
    label="label"
    role="button"
    style="margin: 1rem; padding: 0.5rem; min-width: 80px; border-radius: 4px; display: inline-block; text-align: center; text-transform: uppercase; font-weight: bold; background: rgb(205, 205, 205);"
  >
    label
  </div>
</div>
`;
